x/sequencer/keeper/sequencer.go

```diff
@@ -31,21 +31,27 @@ func (k Keeper) UpdateSequencer(ctx sdk.Context, sequencer types.Sequencer, oldS
 	b := k.cdc.MustMarshal(&sequencer)
 	store.Set(types.SequencerKey(sequencer.SequencerAddress), b)
 
-	oldKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, oldStatus)
-	store.Delete(oldKey)
-
 	seqByrollappKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, sequencer.Status)
 	store.Set(seqByrollappKey, b)
 
-	//handle unbonding queue changes due to status change
-	if sequencer.Status == oldStatus {
-		return
+	//status changed, need to remove old status key
+	if sequencer.Status != oldStatus {
+		oldKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, oldStatus)
+		store.Delete(oldKey)
 	}
-	if sequencer.Status == types.Unbonding {
-		k.setUnbondingSequencerQueue(ctx, sequencer)
-	} else if oldStatus == types.Unbonding {
-		k.removeUnbondingSequencer(ctx, sequencer)
+}
+
+// rotate proposer. we set the first bonded sequencer to be the proposer
+func (k Keeper) RotateProposer(ctx sdk.Context, rollappId string) {
+	seqsByRollapp := k.GetSequencersByRollappByStatus(ctx, rollappId, types.Bonded)
+	if len(seqsByRollapp) == 0 {
+		k.Logger(ctx).Error("no bonded sequencer found for rollapp", "rollappId", rollappId)
```

Comment:
not sure it's an error. as I see it's a valid world state.

Reply:
right. will change in upcoming PR